Code/GWFrames/Extrapolation.py

```diff
@@ -49,6 +49,24 @@ def ValidateSingleWaveform(h5file, filename, WaveformName, ExpectedNModes, Expec
     from re import compile as re_compile
     CompiledModeRegex = re_compile(ModeRegex)
     Valid = True
+
+    # Check identical time columns
+    it = 0
+    Ylm_mode_for_test = sorted(list(h5file[WaveformName]),reverse=True)[it]
+    while Ylm_mode_for_test[0:2] != 'Y_':
+        it += 1
+        Ylm_mode_for_test = sorted(list(h5file[WaveformName]),reverse=True)[it]
+        if it+2 > len(list(h5file[WaveformName])):
+            Valid = False
+            print("FAILED: No modes named 'Y_l*_m*.dat' in "+WaveformName+" in h5 file")
+            break
+    if Valid == True:
+        times = h5file[WaveformName + '/' + Ylm_mode_for_test][:, 0]
+        for mode in h5file[WaveformName]:
+            if mode[0:2] == 'Y_':
+                if not (times == h5file[WaveformName + '/' + mode][:, 0]).all():
+                    Valid = False
+                    print("FAILED: time column for "+mode+" in "+WaveformName+" does not match the others in "+filename+".")
```

Comment:
After looking this over again, I'm think we should just remove this test entirely. Firstly, the test on lines 92-97 (or 74-79 in the original code) checks to make sure that each dataset has the same shape, which can at least rule out the chance the there is a time step that wasn't written to file in a dataset (though it would still pass if the step was just set to zero). And secondly, I have this test written up separately as one of the [quality assurance](https://github.com/10220/quality_assurance) tests, since waveforms should have this test regardless of extrapolation. 

If we do this test on the extracted waveforms, then again when we go to extrapolate them that is just extraneous calculation unless you want this test here for people that are using waveforms from other codes. What are your thoughts on this?